lib/sparkPostTransport.js

```diff
@@ -23,36 +23,130 @@ var SparkPostTransport = function SparkPostTransport(options) {
   return this;
 };
 
-SparkPostTransport.prototype.send = function send(payload, callback) {
-  var email = {
-    transmissionBody: {}
-  };
-
-  // Apply default options and override if provided in mail object
-  email.transmissionBody.tags              = (payload.data.tags) ? payload.data.tags : this.tags;
-  email.transmissionBody.campaign_id       = (payload.data.campaign_id) ? payload.data.campaign_id : this.campaign_id;
-  email.transmissionBody.metadata          = (payload.data.metadata) ? payload.data.metadata : this.metadata;
-  email.transmissionBody.substitution_data = (payload.data.substitution_data) ? payload.data.substitution_data : this.substitution_data;
-  email.transmissionBody.options           = (payload.data.options) ? payload.data.options : this.options;
-  email.transmissionBody.content           = (payload.data.content) ? payload.data.content : this.content;
-  email.transmissionBody.recipients        = (payload.data.recipients) ? payload.data.recipients : this.recipients;
-
-  // Send the transmission using Sparkpost
-  this.sparkPostEmailClient.transmissions.send(email, function( err, res ) {
-    if( err ) {
-      return callback( err );
-    } else {
-      // Example successful Sparkpost transmission response:
-      // { "results": { "total_rejected_recipients": 0, "total_accepted_recipients": 1, "id": "66123596945797072" } }
-      return callback( null, {
-        messageId: res.body.results.id,
-        accepted: res.body.results.total_accepted_recipients,
-        rejected: res.body.results.total_rejected_recipients
-      });
+SparkPostTransport.prototype.send = function send(message, callback) {
+  var self = this
+    , data = message.data
+    , request = {
+      content: {}
     }
+    , customFields = ['tags', 'campaign_id', 'metadata', 'substitution_data', 'options', 'content', 'recipients'];
+
+  // Apply default SP-centric options and override if provided in mail object
+  customFields.forEach(function(fld) {
+    if (data.hasOwnProperty(fld)) {
+      request[fld] = data[fld];
+    } else if (self.hasOwnProperty(fld)) {
+      request[fld] = self[fld];
+    }
+  });
+
+  // Conventional nodemailer fields override SparkPost-specific ones and defaults
+
+  if (data.to) {
+    request.recipients = emailList(data.to) || [];
+  }
+
+  var resolveme = [];
```

Comment:
what do you think about making this a map instead of an array? That might make the srckey dstkey mapping a little easier to read in the resolveAndSend method.